dotCMS/src/integration-test/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImplIntegrationTest.java

```diff
@@ -0,0 +1,130 @@
+        final User user = new UserDataGen().roles(role, backEndUserRole).nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Folder folder1 = new FolderDataGen().site(host).nextPersisted();
+        final Contentlet fileAsset1 = createFileAsset(folder1, "text1", ".txt");
+        final Contentlet fileAsset2 = createFileAsset(folder1, "text2", ".txt");
+
+        final Folder folder2 = new FolderDataGen().site(host).nextPersisted();
+        final Contentlet fileAsset3 = createFileAsset(folder2, "text1", ".txt");
+        final Contentlet fileAsset4 = createFileAsset(folder2, "text2", ".txt");
+
+        this.addPermission(role, folder1, folder2, fileAsset1, fileAsset2, fileAsset3, fileAsset4);
+        final List<FileAsset> files = fileAssetAPIImpl.findFileAssetsByFolder(folder1, user, false);
+
+        assertEquals(2, files.size());
+
+        final List<String> filesInodes = files.stream().map((file) -> file.getInode()).collect(Collectors.toList());
+
+        assertTrue(filesInodes.contains(fileAsset1.getInode()));
+        assertTrue(filesInodes.contains(fileAsset2.getInode()));
+    }
+
+    /**
+     * Method to test: {@link FileAssetAPIImpl#findFileAssetsByFolder(Folder, User, boolean)}
+     * When: Create one Folder with one files, and try to get the files from the first folder with a user without permission
+     * Should: return a empty List
+     *
+     * @throws IOException
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    @Test
+    public void userWithoutPermission() throws IOException, DotDataException, DotSecurityException {
+        final Role backEndUserRole = APILocator.getRoleAPI().loadBackEndUserRole();
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role, backEndUserRole).nextPersisted();
+
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final Folder folder1 = new FolderDataGen().site(host).nextPersisted();
+        createFileAsset(folder1, "text1", ".txt");
+
+        final List<FileAsset> fileAssetsByFolder = fileAssetAPIImpl.findFileAssetsByFolder(folder1, user, false);
+
+        assertTrue(fileAssetsByFolder.isEmpty());
+    }
+
+    private Contentlet createFileAsset(
+            final Folder folder,
+            final String fileName,
+            final String suffix)
+            throws IOException, DotSecurityException, DotDataException {
+        final java.io.File file = java.io.File.createTempFile(fileName, suffix);
+        FileUtil.write(file, "helloworld");
+
+        return  new FileAssetDataGen(folder, file).nextPersisted();
+    }
+
+    private void addPermission(
+            final Role role,
+            final Permissionable... permissionables)
+            throws DotDataException, DotSecurityException {
+
+        final User systemUser = APILocator.systemUser();
+
+        for (Permissionable permissionable : permissionables) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'permissionable' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5280157)